src/atoms/Switch/Switch.native.js

```diff
@@ -0,0 +1,315 @@
+    KNOB_TRANSLATION_DURATION: 200,
+    KNOB_SCALE_DURATION: 100,
+  };
+
+  const leftSpace = useRef(new Animated.Value(SWITCH_STATE_CONFIGS.OFF_LEFT_SPACE));
+  const rightSpace = useRef(new Animated.Value(SWITCH_STATE_CONFIGS.OFF_RIGHT_SPACE));
+
+  useEffect(() => {
+    if (isPropDefined(on)) {
+      if (isPropDefined(defaultOn)) {
+        throw Error('Expected only one of defaultOn or on props.');
+      }
+      if (on && !toggleState) {
+        moveKnob(
+          [
+            {
+              animationProp: leftSpace.current,
+              toValue: SWITCH_STATE_CONFIGS.ON_LEFT_SPACE,
+              duration: SWITCH_STATE_CONFIGS.KNOB_TRANSLATION_DURATION,
+            },
+            {
+              animationProp: rightSpace.current,
+              toValue: SWITCH_STATE_CONFIGS.ON_RIGHT_SPACE,
+              duration: SWITCH_STATE_CONFIGS.KNOB_TRANSLATION_DURATION,
+            },
+          ],
+          () => {
+            setToggleState(!toggleState);
+          },
+        );
+      } else if (!on && toggleState) {
+        moveKnob(
+          [
+            {
+              animationProp: leftSpace.current,
+              toValue: SWITCH_STATE_CONFIGS.OFF_LEFT_SPACE,
+              duration: SWITCH_STATE_CONFIGS.KNOB_TRANSLATION_DURATION,
+            },
+            {
+              animationProp: rightSpace.current,
+              toValue: SWITCH_STATE_CONFIGS.OFF_RIGHT_SPACE,
+              duration: SWITCH_STATE_CONFIGS.KNOB_TRANSLATION_DURATION,
+            },
+          ],
+          () => {
+            setToggleState(!toggleState);
+          },
+        );
+      }
+    }
+    if (defaultOn) {
+      leftSpace.current.setValue(SWITCH_STATE_CONFIGS.ON_LEFT_SPACE);
+      rightSpace.current.setValue(SWITCH_STATE_CONFIGS.ON_RIGHT_SPACE);
+      setToggleState(true);
+    }
+  }, [defaultOn, on]);
+
+  const onPressIn = () => {
+    if (toggleState) {
+      Animated.timing(leftSpace.current, {
+        toValue: SWITCH_STATE_CONFIGS.ON_ACTIVE_LEFT_SPACE,
+        duration: SWITCH_STATE_CONFIGS.KNOB_SCALE_DURATION,
+      }).start();
+    } else {
+      Animated.timing(rightSpace.current, {
+        toValue: SWITCH_STATE_CONFIGS.OFF_ACTIVE_RIGHT_SPACE,
+        duration: SWITCH_STATE_CONFIGS.KNOB_SCALE_DURATION,
+      }).start();
+    }
+    setActive(true);
+  };
+
+  const onPressOut = () => {
+    setActive(false);
+    if (isPropDefined(on)) {
+      if (toggleState) {
+        Animated.timing(leftSpace.current, {
+          toValue: SWITCH_STATE_CONFIGS.ON_LEFT_SPACE,
+          duration: SWITCH_STATE_CONFIGS.KNOB_SCALE_DURATION,
+        }).start(() => {
+          onChange(!on);
+        });
+      } else {
+        Animated.timing(rightSpace.current, {
+          toValue: SWITCH_STATE_CONFIGS.OFF_RIGHT_SPACE,
+          duration: SWITCH_STATE_CONFIGS.KNOB_SCALE_DURATION,
+        }).start(() => {
+          onChange(!on);
+        });
+      }
+      return;
+    }
+    if (toggleState) {
+      moveKnob(
```

Comment:
This seems like an unnecessary abstraction. Can't we simply have two functions called ```moveKnobLeft``` & ```moveKnobRight```? 

Here we have a function that accepts an array of configs and a callback. It's just harder to understand and refactor. Having two simple functions that do exactly what you see in the animation makes more sense (move left & move right).